pennylane_lightning/lightning_qubit/lightning_qubit.py

```diff
@@ -301,25 +295,32 @@ def create_ops_list(self):
         @property
         def measurements(self):
             """Returns a Measurements object matching ``use_csingle`` precision."""
-            ket = np.ravel(self._state)
-            state_vector = StateVectorC64(ket) if self.use_csingle else StateVectorC128(ket)
             return (
-                MeasurementsC64(state_vector)
+                MeasurementsC64(self.state_vector)
                 if self.use_csingle
-                else MeasurementsC128(state_vector)
+                else MeasurementsC128(self.state_vector)
             )
 
         @property
         def state(self):
-            """Returns the flattened state vector."""
-            shape = (1 << self.num_wires,)
-            return self._reshape(self._pre_rotated_state, shape)
+            """Copy the state vector data to a numpy array.
+
+            **Example**
+
+            >>> dev = qml.device('lightning.kokkos', wires=1)
+            >>> dev.apply([qml.PauliX(wires=[0])])
+            >>> print(dev.state)
+            [0.+0.j 1.+0.j]
+            """
+            state = np.zeros(2**self.num_wires, dtype=self.C_DTYPE)
+            state = self._asarray(state, dtype=self.C_DTYPE)
+            self._qubit_state.getState(state.ravel(order="C"))
```

Comment:
Do we need the ravel and order call here?

Reply:
Not needed, this was a no-op in this case though. Removed.